cmake/compiler-flags.cmake

```diff
@@ -23,10 +23,10 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 include(debug-info)
 
-if(CMAKE_BUILD_TYPE STREQUAL "RELEASE")
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
Note that this is not correct fix, this will now errneously not match `rELEASE` string

Reply:
ok, I will check tomorrow. Thanks for the feedback.

Reply:
Hmm, how come you say `rELEASE` is a valid input? Looking at https://cmake.org/cmake/help/v3.0/variable/CMAKE_BUILD_TYPE.html:
> Possible values are empty, Debug, Release, RelWithDebInfo and MinSizeRel.

Reply:
case-insensitivity is how it is currently handled, it should stay that way be it either via
the LHS of the diff, or by fixing the every CMAKE_BUILD_TYPE-dependent check
to be case-insensitive.

Reply:
ok. please read my last commit message. If you persist on making it case-insensitive I will see how I can achieve that goal.